connector-email/src/test/java/com/redhat/cloud/notifications/connector/email/RecipientsListTest.java

```diff
@@ -0,0 +1,141 @@
+
+@QuarkusTest
+@QuarkusTestResource(TestLifecycleManager.class)
+public class RecipientsListTest extends CamelQuarkusTestSupport {
+
+    @Inject
+    EmailConnectorConfig emailConnectorConfig;
+
+    @Inject
+    ProducerTemplate producerTemplate;
+
+    @Inject
+    ObjectMapper objectMapper;
+
+    @Override
+    public boolean isUseRouteBuilder() {
+        return false;
+    }
+
+    Exchange test(ExpectationResponseCallback verifyEmptyRequest) throws Exception {
+
+        getMockHttpRequest("/internal/recipients-resolver", "PUT", verifyEmptyRequest);
+
+        Exchange exchange = createExchangeWithBody(new HashSet<>());
+        exchange.setProperty(EMAIL_RECIPIENTS, new HashSet<>());
+        exchange.setProperty(RECIPIENT_SETTINGS, new ArrayList<>());
+        exchange.setProperty(START_TIME, System.currentTimeMillis());
+
+        adviceWith(SPLIT_AND_SEND, context(), new AdviceWithRouteBuilder() {
+            @Override
+            public void configure() throws Exception {
+                mockEndpointsAndSkip(
+                    "direct:" + SEND_EMAIL_BOP
+                );
+            }
+        });
+
+        adviceWith(emailConnectorConfig.getConnectorName(), context(), new AdviceWithRouteBuilder() {
+            @Override
+            public void configure() throws Exception {
+                mockEndpoints(
+                    "direct:" + SUCCESS,
+                            "direct:" + SPLIT_AND_SEND
+                );
+            }
+        });
+
+        return exchange;
+
+    }
+
+    @Test
+    void testEmpty() throws Exception {
+
+        ExpectationResponseCallback verifyEmptyRequest = req -> response().withBody("[]").withStatusCode(200);
+
+        Exchange exchange = test(verifyEmptyRequest);
+        MockEndpoint successEndpoint = getMockEndpoint("mock:direct:" + SUCCESS);
+        MockEndpoint splitRoute = getMockEndpoint("mock:direct:" + SPLIT_AND_SEND);
+
+        splitRoute.expectedMessageCount(0);
+        successEndpoint.expectedMessageCount(1);
+
+        producerTemplate.send("seda:" + ENGINE_TO_CONNECTOR, exchange);
+        splitRoute.assertIsSatisfied();
+        successEndpoint.assertIsSatisfied();
+    }
+
+    @Test
+    void testNotEmpty() throws Exception {
+        Set<User> users = TestUtils.createUsers("user-1", "user-2", "user-3", "user-4", "user-5", "user-6", "user-7");
+        String strUsers = objectMapper.writeValueAsString(users);
+        ExpectationResponseCallback verifyEmptyRequest = req -> response().withBody(strUsers).withStatusCode(200);
+
+        Exchange exchange = test(verifyEmptyRequest);
+        MockEndpoint successEndpoint = getMockEndpoint("mock:direct:" + SUCCESS);
+        MockEndpoint splitRoute = getMockEndpoint("mock:direct:" + SPLIT_AND_SEND);
+        MockEndpoint bopRoute = getMockEndpoint("mock:direct:" + SEND_EMAIL_BOP);
+
+        splitRoute.expectedMessageCount(1);
+        successEndpoint.expectedMessageCount(1);
```

Comment:
Sure, we expect 1 message on success endpoint. This `expectedMessageCount(.)` works actually as a minimum message count.